src/connection.rs

```diff
@@ -818,11 +818,18 @@ impl ConnectionLike for Connection {
         self.con.send_bytes(cmd)?;
         let mut rv = vec![];
         for idx in 0..(offset + count) {
-            let item = self.read_response()?;
-            if idx >= offset {
-                rv.push(item);
+            // When processing a transaction, some responses may be errors.
+            // We need to keep processing the rest of the responses in that case,
+            // so bailing early with `?` would not be correct.
+            // See: https://github.com/mitsuhiko/redis-rs/issues/436
+            let response = self.read_response();
```

Comment:
Any errors occuring will just be lost here, I'd expect an error (probably the first?) to still be returned after consuming the remaining commands.

Reply:
@Marwes the error from the failed transaction will be returned (`Transaction discarded because of previous errors.`). I thought of returning a vector of `Results`, but that would change the API quite a bit.

Reply:
`Transaction discarded because of previous errors.` is a pretty bad error though. Chances are the first error that occured is the one with the most information returning that seems to make the most sense to me.

Reply:
Very well, I'll proceed in that direction.

Reply:
I changed the code as suggested -- the first error is now returned.